dm/simulator/internal/mcp/mcp.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package mcp defines the Modification Candidate Pool (MCP).
+package mcp
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"go.uber.org/zap"
+)
+
+var (
+	initRandSeedOnce sync.Once
+	gMCPRand         *rand.Rand
+	gMCPRandLock     sync.Mutex
+)
+
+func init() {
```

Comment:
this function is only called once by go runtime, no need to use a `sync.Once` I think

And `var gMCPRand = rand.New(rand.NewSource(time.Now().Unix()))` should have same effect, no need to explicitly use an `init()`